src/Symfony/Component/Messenger/Transport/Enhancers/StopWhenTimeLimitIsReachedReceiver.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Transport\Enhancers;
+
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Messenger\Transport\ReceiverInterface;
+
+/**
+ * @author Simon Delicata <simon.delicata@free.fr>
+ */
+class StopWhenTimeLimitIsReachedReceiver implements ReceiverInterface
+{
+    private $decoratedReceiver;
+    private $timeLimit;
+    private $logger;
+
+    public function __construct(ReceiverInterface $decoratedReceiver, int $timeLimit, LoggerInterface $logger = null)
```

Comment:
please name the variable like $timeLimitInSeconds to make it clear what unit this is in